org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService.xtend

```diff
@@ -100,8 +100,14 @@ class RenameService implements IRenameService, IRenameServiceExtension {
 
 	protected def EObject getElementAtOffset(XtextResource xtextResource, Document document, Position caretPosition) {
 		val caretOffset = document.getOffSet(caretPosition)
-		val leafNode = NodeModelUtils.findLeafNodeAtOffset(xtextResource.parseResult.rootNode, caretOffset)
-		val offset = if (caretOffset === xtextResource.parseResult.rootNode.endOffset && caretOffset > 0 && leafNode.offset === caretOffset && !isIdentifier(leafNode)) 
+		val rootNode = xtextResource?.parseResult?.rootNode
+		if (rootNode === null)
+			return null
+		val leafNode = NodeModelUtils.findLeafNodeAtOffset(rootNode, caretOffset)
+		val offset = if (caretOffset > 0 &&
+			(caretOffset === rootNode.totalEndOffset 
+				|| leafNode === null 
+				|| (leafNode.offset === caretOffset && !isIdentifier(leafNode))))
 				caretOffset - 1
```

Comment:
My feeling is that a single offset-1 could still yield something that does not fulfill this condition here. Could you also give it a name, eg extract a boolean method?

Reply:
Sorry you feel this way :-) I can stand corrections, so if you know something _is_ wrong, please state it. Otherwise, this tone is a pretty patronizing.

I changed it completely anyway.